qa327_test/backend/login/test_backend_login.py

```diff
@@ -1 +1,35 @@
-# All tests for R1 - /login go here
\ No newline at end of file
+import pytest
+import requests
+
+from qa327_test.conftest import base_url
+import qa327.backend as bn
+from qa327.models import User
+
+# Mock a sample user
+test_user = User(
+    email='test@test.com',
+    name='Test User',
+    password='Password1!'
+)
+
+@pytest.mark.usefixtures('server')
+def test_backend_login():
+    # Set up valid user account
+    bn.register_user(test_user.email, test_user.name, test_user.password, test_user.password)
+
+    # Test input partition #1 (valid email, correct password)
+    result = bn.login_user(test_user.email, test_user.password)
+    assert result is not None
+    assert result.name == test_user.name
+
+    # Test input partition #2 (valid email, incorrect password)
+    result = bn.login_user(test_user.email, "IncorrectPassword1!")
+    assert result == None
+
+    # Test input partition #3 (invalid email, correct password)
+    result = bn.login_user("incorrect@email.com", test_user.password)
+    assert result == None
+
+    # Test input partition #4 (invalid email, incorrect password)
+    result = bn.login_user("incorrect@email.com", "IncorrectPassword1!")
+    assert result == None
```

Comment:
Looks good and simple enough to me. Ready to merge.

I'll throw in one thing though, you can take it or leave it. The reason is, I'm not sure if this change I propose will result in that much overall benefit since the case here is pretty simple.

There are multiple test cases in this one "test case" `test_backend_login()`. It doesn't look like that much of a problem since there are only 4 here. However, if we continue with this model and add more tests, or if another feature (ie sell backend) has many partitions, then it would be better practice to make classes and throw in each partition as a function of the class
```
import unittest 


@pytest.mark.usefixtures('server')
class Test(unittest.Testcase):
  def test_partition_one(self):
    ...
  def test_partition_two(self)
    ...
```
This way, `pytest` still runs as it does now, plus the added advantage of `unittest` telling us exactly which function aka partition test case, is failing.
As I said, it's a minor fix because our cases are pretty simple. This just may be better practice - if you want to exercise that option.

Reply:
I also found [this](https://docs.pytest.org/en/3.0.1/unittest.html) helpful to refer to

Reply:
Good point - I'll separate them into different functions. I should probably keep using `pytest` though, I don't see a need to combine two different testing frameworks.